app/questionnaire/questionnaire_schema.py

```diff
@@ -99,11 +103,64 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_for_numeric_answers()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    @cached_property
+    def min_and_max_map(self) -> ImmutableDict[str, dict[str, int]]:
+        return ImmutableDict(self._min_and_max_map)
+
+    def _create_min_max_map(
+        self,
+        min_max: Any,
+        answer_id: str,
+        answers: list,
+        default_min_max: str,
+    ) -> None:
+        longest_string: int = 0
+        for answer in answers:
+            value = answer.get(min_max, {}).get("value")
+
+            if isinstance(value, int):
+                if longest_string and len(str(value)) > longest_string:
+                    longest_string = len(str(value))
+                if not longest_string:
+                    longest_string = len(str(value))
+
+            elif isinstance(value, dict) and value:
+                if value.get("source") == "answers":
+                    if longest_string != 0 and (
+                        int(self._min_and_max_map[value["identifier"]][min_max])
+                        > longest_string
+                    ):
+                        longest_string = int(
+                            self._min_and_max_map[value["identifier"]][min_max]
+                        )
+                    if longest_string == 0:
+                        longest_string = int(
+                            self._min_and_max_map[value["identifier"]][min_max]
+                        )
+
+        if longest_string:
+            self._min_and_max_map[answer_id][min_max] = longest_string  # type: ignore
+        else:
```

Comment:
could simplify to `self._min_and_max_map[answer_id][min_max] = longest_string or len(default_min_max)`